### ISSUE TITLE
Failed to send the report only during hot-plugged event 

### ISSUE THREAD

**Issue by JunzheFan** [WICG/webhid#111]
I have been tested my device which is a headset connector with WebHID for a few days, I encountered a strange behaviour where I can not send the report using the device object received on the hot-plugged event, it throwed a DOMException with permisson not allowed.  I tried a few workaround and found that if I set a 3 seconds delay then acquired the device, the output report can be successfully sent to device.  Could it be the device is not actually ready for operation when the connect event fired ?   Below is code i used to showcase the two situations mentioned above, thanks in advance  

```
    navigator.hid.addEventListener("connect", async ({ device }) => {
      // Failed to write the report 
      // if (!device.opened) {
      //   await device.open();
      // }
      // const data = new Uint8Array([0x6, 0x1, 0x0, 0, 0, 0, 0, 0]);
      // device.sendReport(data[0], data.slice(1, data.length));
      
      // if set a 2-3 seconds delay, then get the device, the report can be successfully sent 
      setTimeout(async () => {
        let [Device] = await navigator.hid.getDevices();
        if (!Device.opened) {
          await Device.open();
        }
        const data = new Uint8Array([0x6, 0x1, 0x0, 0, 0, 0, 0, 0]);
        Device.sendReport(data[0], data.slice(1, data.length));
      }, 3000);
    });
```

**Comment by nondebug** [comment]
Is anything logged in chrome://device-log or the developer console?

I agree this is strange. I don't think this is an implementation bug, this seems like a device error. If it were related to Chrome's permission model then it should have failed to open the device, but based on your description it sounds like open succeeded but sendReport failed.

**Comment by JunzheFan** [comment]
yes you are correct, then device can be successfully open, I went to the chrome://device-log and got the following error message,  it seems strange that it's saying the device does not support output report while I can succesully write except for the hot-plugged event,  maybe the device is not actually ready when the Window OS detected the device like it needs a few seconds to do the setup, is there a solution to know whether the device is ready for operation ? 
![WXWorkCapture_16865342256653](https://github.com/WICG/webhid/assets/87253480/a6c9cbaf-ca45-48dd-bcc5-fcac428647cf)

**Comment by JunzheFan** [comment]
@nondebug I am not quite understand this bit of the documentation, so in my case my device has three sub hid devices, does that mean the **connect** event will fire three times ? for now, it seems only firing once. thanks 
![WXWorkCapture_16865364698881](https://github.com/WICG/webhid/assets/87253480/ecb4282a-7fc6-4c09-a774-1c43035f04b5) 
here is my device structure 
![WXWorkCapture_16865371835876](https://github.com/WICG/webhid/assets/87253480/e3e3fc1d-4477-4023-8ad2-b3b9ddc247aa)

**Comment by JunzheFan** [comment]
@nondebug HI matt, I found that my problem is essentially a duplicate of this issue reported two years ago,  https://github.com/WICG/webhid/issues/65, since on Windows it seems not possible to wait for all collections to be enumerated before firing the connect event, is there any update on the implementation of the  'devicechange' event you mentioned as sort of workaround in the issue
